File: desktop/src/utils/cardLabels.test.js
import test from 'node:test';
import assert from 'node:assert/strict';
import { readFileSync } from 'node:fs';
import { ATTRIBUTE_LABELS, RACE_LABELS, attributeLabel, raceLabel, typeGroup } from './cardLabels.js';

// ZWILLING: android CardLabelsTest.kt liest dieselbe Fixture.
const F = JSON.parse(readFileSync(new URL('../../../docs/fixtures/valuation/card-labels.json', import.meta.url), 'utf8'));

test('Kartenart-Gruppen (Fixture)', () => {
  for (const c of F.typeGroups) assert.equal(typeGroup(c.type), c.group, String(c.type));
});

test('Attribute und Monstertypen deutsch, Unbekanntes bleibt (Fixture)', () => {
  assert.deepEqual(ATTRIBUTE_LABELS, F.attributes);
  assert.deepEqual(RACE_LABELS, F.races);
  for (const [en, de] of Object.entries(F.attributes)) assert.equal(attributeLabel(en), de);
  for (const [en, de] of Object.entries(F.races)) assert.equal(raceLabel(en), de);
  for (const x of F.passThrough) { assert.equal(attributeLabel(x), x); assert.equal(raceLabel(x), x); }
});

File: android/app/src/test/java/com/example/yugiohscanner/CardLabelsTest.kt
package com.example.yugiohscanner

import com.example.yugiohscanner.DeckFixtureWorld.Companion.objects
import com.example.yugiohscanner.ml.CardLabels
import org.json.JSONObject
import org.junit.Assert.assertEquals
import org.junit.Test

/** ZWILLING von desktop/src/utils/cardLabels.test.js -- dieselbe Fixture docs/fixtures/valuation/card-labels.json. */
class CardLabelsTest {
    private val f = JSONObject(Fixtures.text("docs/fixtures/valuation/card-labels.json"))
    private fun JSONObject.toMap() = keys().asSequence().associateWith { getString(it) }

    @Test fun `Kartenart-Gruppen`() {
        for (c in f.getJSONArray("typeGroups").objects()) {
            val t = if (c.isNull("type")) null else c.getString("type")
            assertEquals(t.toString(), c.getString("group"), CardLabels.typeGroup(t))
        }
    }

    @Test fun `Attribute und Monstertypen deutsch, Unbekanntes bleibt`() {
        assertEquals(f.getJSONObject("attributes").toMap(), CardLabels.ATTRIBUTES)
        assertEquals(f.getJSONObject("races").toMap(), CardLabels.RACES)
        val pass = f.getJSONArray("passThrough")
        for (i in 0 until pass.length()) {
            val x = pass.getString(i)
            assertEquals(x, CardLabels.attribute(x))
            assertEquals(x, CardLabels.race(x))
        }
    }
}
